CesiumIonClient/src/Connection.cpp

```diff
@@ -582,6 +588,85 @@ Defaults defaultsFromJson(const rapidjson::Document& json) {
   return defaults;
 }
 
+GeocoderResult geocoderResultFromJson(const rapidjson::Document& json) {
+  GeocoderResult result;
+
+  auto featuresMemberIt = json.FindMember("features");
+  if (featuresMemberIt != json.MemberEnd() &&
+      featuresMemberIt->value.IsArray()) {
+    auto featuresIt = featuresMemberIt->value.GetArray();
+    for (auto& feature : featuresIt) {
+      const rapidjson::Value* pLabel =
+          rapidjson::Pointer("/properties/label").Get(feature);
+      if (!pLabel) {
+        SPDLOG_WARN("Missing label for geocoder feature");
+        continue;
+      }
+
+      std::string label(pLabel->GetString());
+      auto bboxMemberIt = feature.FindMember("bbox");
+      if (bboxMemberIt == feature.MemberEnd() ||
+          !bboxMemberIt->value.IsArray()) {
+        // Could be a point value.
+        const rapidjson::Value* pCoordinates =
+            rapidjson::Pointer("/geometry/coordinates").Get(feature);
+        if (!pCoordinates) {
+          SPDLOG_WARN(
+              "Missing bbox and geometry.coordinates for geocoder feature");
+          continue;
+        }
+
+        if (!pCoordinates->IsArray() || pCoordinates->Size() != 2) {
+          SPDLOG_WARN("geometry.coordinates must be an array of size 2");
+          continue;
+        }
+
+        auto coordinatesArray = pCoordinates->GetArray();
+
+        CesiumGeospatial::Cartographic point =
+            CesiumGeospatial::Cartographic::fromDegrees(
+                JsonHelpers::getDoubleOrDefault(coordinatesArray[0], 0),
+                JsonHelpers::getDoubleOrDefault(coordinatesArray[1], 0));
+
+        result.features.emplace_back(label, point);
+      } else {
+        auto bboxIt = bboxMemberIt->value.GetArray();
+        if (bboxIt.Size() != 4) {
+          SPDLOG_WARN("bbox property should have exactly four values");
+          continue;
+        }
+
+        CesiumGeospatial::GlobeRectangle rect =
+            CesiumGeospatial::GlobeRectangle::fromDegrees(
+                JsonHelpers::getDoubleOrDefault(bboxIt[0], 0),
+                JsonHelpers::getDoubleOrDefault(bboxIt[1], 0),
+                JsonHelpers::getDoubleOrDefault(bboxIt[2], 0),
+                JsonHelpers::getDoubleOrDefault(bboxIt[3], 0));
+
+        result.features.emplace_back(label, rect);
+      }
+    }
+  }
+
+  auto attributionMemberIt = json.FindMember("attributions");
+  if (attributionMemberIt != json.MemberEnd() &&
+      attributionMemberIt->value.IsArray()) {
+    const auto& valueJson = attributionMemberIt->value;
+
+    result.attributions.reserve(valueJson.Size());
+
+    for (rapidjson::SizeType i = 0; i < valueJson.Size(); ++i) {
+      const auto& element = valueJson[i];
```

Comment:
It's ok to use `auto` for iterators, but this is more clear without:
```suggestion
      const rapidjson::Value& element = valueJson[i];
```